mock/src/store.rs

```diff
@@ -191,6 +191,41 @@ impl Store for MockStore {
                             .push(EntityChange::from_key(key, EntityChangeOperation::Set));
                     }
                 }
+                EntityOperation::Update { key, data, guard } => {
+                    let entities_of_type = entities
+                        .entry(key.subgraph_id.clone())
+                        .or_default()
+                        .entry(key.entity_type.clone())
+                        .or_default();
+
+                    if entities_of_type.contains_key(&key.entity_id) {
+                        let existing_entity = entities_of_type.get_mut(&key.entity_id).unwrap();
+                        if let Some(filter) = guard {
+                            if !entity_matches_filter(existing_entity, &filter) {
+                                return Err(TransactionAbortError::AbortUnless {
+                                    expected_entity_ids: vec![key.entity_id],
+                                    actual_entity_ids: vec![],
+                                    description:
+                                        "update failed because entity does not match guard"
+                                            .to_owned(),
+                                }
+                                .into());
+                            }
+                        }
+                        // FIXME: Check that existing_entity matches guard
```

Comment:
Is this something that still needs to be done?

Reply:
Ugh. No, I added the checking at some point when I was working on it, and forgot to take out the comment. Will do that and update the PR.